src/context.ts

```diff
@@ -50,9 +50,21 @@ function createContext<TState extends State>(_initialState?: TState) {
     )
   }
 
+  const useStoreApi = () => {
+    // ZustandContext value is guaranteed to be stable.
+    const useProviderStore = useContext(ZustandContext)
+    if (!useProviderStore) {
+      throw new Error(
+        'Seems like you have not used zustand provider as an ancestor.'
+      )
+    }
+    return useProviderStore
```

Comment:
Oops, sorry, I should have explained better.
This is exactly the pattern we want to avoid. Like, this is the whole point of this utility.
(If this were possible, we wouldn't need `useStore`, or even this utility.)

I hope something like this would work?
```suggestion
    return {
      getState: useProviderStore.getState,
      setState: useProviderStore.setState,
      subscribe: useProviderStore.subscribe,
    }
```

Reply:
Ah okay you want to "de-functionalize" it, I suppose? What about the `destroy` method?

Reply:
> Ah okay you want to "de-functionalize" it, I suppose?

Yes. It would avoid misusage. (Otherwise, people can violate rules of hooks and eslint can't detect it.)

> What about the `destroy` method?

Oh, I missed it. But, should we add it? I don't see any use cases, but adding it is fine.